src/ScriptEditor/ScriptEditor.ts

```diff
@@ -63,26 +63,44 @@ export class ScriptEditor {
 
     // Add ts definitions for API
     const source = netscriptDefinitions.replace(/export /g, "");
-    for (const languageDefaults of [
-      monaco.languages.typescript.javascriptDefaults,
-      monaco.languages.typescript.typescriptDefaults,
-    ]) {
+    for (const [language, languageDefaults] of [
+      ["javascript", monaco.languages.typescript.javascriptDefaults],
+      ["typescript", monaco.languages.typescript.typescriptDefaults],
+    ] as ["javascript" | "typescript", monaco.languages.typescript.LanguageServiceDefaults][]) {
```

Comment:
Oof, this is a particular ugly cast. Why is it needed? Can you just declare `language` as a string instead?

Reply:
Without any cast, it deduces the type of the array as
```typescript
(string | monaco.languages.typescript.LanguageServiceDefaults)[]
```
instead of an array of tuples.

I definitely could just use `string` for `language`, but this will catch typos in tests against language.

Reply:
What if you iterate over
```js
for (const {language, languageDefaults} of [
      {language: "javascript", languageDefaults: monaco.languages.typescript.javascriptDefaults},
      {language: "typescript", languageDefaults: monaco.languages.typescript.typescriptDefaults},
]) {
```
? Based on playground, that will require no extra type annotations, and is very clear.

Reply:
Actually, `as const` works, and gives a narrow bound to `language` (specifically, it gives it the type `"javascript" | "typescript"`).